lms/djangoapps/certificates/models.py

```diff
@@ -574,3 +579,92 @@ def get_config(cls):
         instance = cls.current()
         json_data = json.loads(instance.configuration) if instance.enabled else {}
         return json_data
+
+
+class BadgeAssertion(models.Model):
+    """
+    Tracks badges on our side of the badge baking transaction
+    """
+    user = models.ForeignKey(User)
+    course_id = CourseKeyField(max_length=255, blank=True, default=None)
+    data = JSONField()
+
+    @property
+    def image_url(self):
+        """
+        Get the image for this assertion.
+        """
+        return self.data['json']['image']
+
+    class Meta(object):
+        """
+        Meta information for Django's construction of the model.
+        """
+        unique_together = (('course_id', 'user'),)
+
+
+class BadgeImageConfiguration(models.Model):
+    """
+    Contains the configuration for badges for a specific mode. The mode
+    """
+    mode = models.CharField(
+        max_length=125,
+        help_text=_(u"The course mode this image is for, like 'verified' or 'honor'."),
+        unique=True,
+    )
+    icon = models.ImageField(
+        # Actual max is 256KB, but need overhead for badge baking. This should be more than enough.
+        help_text=_(
+            u"Badge images must be square PNG files. To use with Mozilla Backpack, the filesize should be "
+            u"under 250 kilobytes."
```

Comment:
We talked about validating this on the model, upon upload - since these are the generic images, you can't really create the badges on the badgr server now to fully validate the badge creation of a specific course, but you can check the image type and size. Another option would be to create a sample badge on the server, just for the purpose of validating the image, if you want to fully delegate this to badgr and avoid duplicating the checks.